test/tstRay.cpp

```diff
@@ -375,6 +376,62 @@ BOOST_AUTO_TEST_CASE(ray_sphere_intersection,
 #undef ARBORX_TEST_RAY_SPHERE_INTERSECTION
 #undef ARBORX_TEST_RAY_SPHERE_NO_INTERSECTION
 
+BOOST_AUTO_TEST_CASE(intersects_triangle)
+{
+  using ArborX::Experimental::Ray;
+  using ArborX::Experimental::Triangle;
+  constexpr Triangle unit_triangle{{0, 0, 0}, {1, 0, 0}, {0, 1, 0}};
+
+  BOOST_TEST(intersects(Ray{{.1, .2, .3}, {0, 0, -1}}, unit_triangle));
+  BOOST_TEST(intersects(Ray{{1.1, 1.2, 1}, {-1, -1, -1}}, unit_triangle));
+  BOOST_TEST(intersects(Ray{{-1.9, 3.2, -1}, {2, -3, 1}}, unit_triangle));
+
+  // ray origin on the triangle
```

Comment:
Shall we consider these cases true instead of false? It is a choice, but it is better to be consistent with the results of ray-box intersects (see all the ray box tests under the comment "origin is on the box").